### ISSUE TITLE
Compatibility matrix

### ISSUE THREAD

**Issue by inimaz** [mlco2/codecarbon#469]
### Description
**Goal**: A new user comes with OS `X` + GPU `Y` + CPU `Z`. Is `codecarbon` compatible with her system? If so, how to install it? What are the limitations?

Codecarbon uses different tools to measure each CPU/GPU/hardware. Each of these tools have their own limitations and installation steps. Sometimes they depend on the OS of the machine.

It would be very interesting to have a centralized file explaining to which systems `codecarbon` is compatible. In this file we could explain:
* OS (e.g. Linux)
* tool that measures it (e.g. RAPL)
* specific installation steps
* limitations if any (e.g. https://github.com/mlco2/codecarbon/issues/244)

**Comment by benoit-cty** [comment]
Good Idea, I've started a diagram in https://app.diagrams.net/#Hmlco2%2Fcodecarbon%2Fuse-cpu-load%2Fdocs%2Fedit%2Fcpu_fallback.drawio
![image](https://github.com/mlco2/codecarbon/assets/6603048/8b089911-f85b-43ee-b5b9-71d1cd87cbba)
